### PR TITLE
Https initial playlist

### PR DESCRIPTION
**Pull request by milos-pesic-sc**

### CODE REVIEW

File: include/skippyHLS/skippy_hls.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#pragma once
```
Comment (milos-pesic-sc) [inline include/skippyHLS/skippy_hls.h]: This should be the only exported header now.

File: src/skippy_hls_priv.h [hunk 2]
```diff
@@ -1,12 +1,7 @@
 #pragma once
 
-#include <glib.h>
-
-#define SKIPPY_HLS_DOWNLOAD_AHEAD "skippy-download-ahead"
-#define SKIPPY_HLS_ERROR skippy_hls_error_quark()
-G_GNUC_INTERNAL GQuark skippy_hls_error_quark(void);
-
 typedef enum {
```
Comment (milos-pesic-sc) [inline src/skippy_hls_priv.h]: HlsInternalError codes defined here. These are error codes used internally to represent various errors.

File: src/skippy_hlsdemux.c [hunk 3]
```diff
@@ -541,21 +542,26 @@ skippy_hls_demux_handle_first_playlist (SkippyHLSDemux* demux)
...
-      REPORT_FATAL_ERROR (demux, RESOURCE, READ,  ("First playlist: Incomplete M3U8 data."), ("%s", skippy_m3u8_client_get_current_raw_data (demux->client)));
-    }
-    else {
-      REPORT_FATAL_ERROR (demux, STREAM, DECODE, ("First playlist: Invalid M3U8 data (buffer=%p)", demux->playlist), (NULL));
-    }
-    goto error;
+  switch (result) {
+    case PLAYLIST_INCOMPLETE:
+      GST_OBJECT_UNLOCK (demux);
+      GST_ELEMENT_ERROR (demux, SKIPPY_HLS, PLAYLIST_INCOMPLETE_ON_LOAD, ("First playlist: Incomplete M3U8 data."), ("%s", skippy_m3u8_client_get_current_raw_data (demux->client)));
```
Comment (milos-pesic-sc) [inline src/skippy_hlsdemux.c]: Reporting newly defined error to the rest of the world.

File: src/skippy_hlsdemux.c [hunk 4]
```diff
@@ -541,21 +542,26 @@ skippy_hls_demux_handle_first_playlist (SkippyHLSDemux* demux)
...
+      goto error;
+      break;
+    case PLAYLIST_INVALID_UTF_CONTENT:
+      GST_OBJECT_UNLOCK (demux);
+      GST_ELEMENT_ERROR (demux, SKIPPY_HLS, PLAYLIST_INVALID_UTF_CONTENT, ("First playlist: Invalid M3U8 data (buffer=%p)", demux->playlist), (NULL));
```
Comment (milos-pesic-sc) [inline src/skippy_hlsdemux.c]: Same as above

File: src/skippy_hlsdemux.c [hunk 5]
```diff
@@ -1293,9 +1300,9 @@ void skippy_hls_demux_append_query_param_to_hls_url (gchar **url, const gchar* q
 }
 
 G_GNUC_INTERNAL
-void skippy_hlsdemux_setup (void)
+void skippy_hlsdemux_setup (guint hls_demux_rank)
```
Comment (milos-pesic-sc) [inline src/skippy_hlsdemux.c]: Adding support for setting custom rank. Handy for disabling skippy hlsdemux.

File: include/skippyHLS/skippy_hls.h (hunk 1 above)
Comment (tchakabam) [inline include/skippyHLS/skippy_hls.h]: :+1:

**Comment by pokey909** [comment]
:+1:

File: src/skippy_hlsdemux.c [hunk 6]
```diff
@@ -541,21 +542,26 @@ skippy_hls_demux_handle_first_playlist (SkippyHLSDemux* demux)
...
+      GST_ELEMENT_ERROR (demux, SKIPPY_HLS, PLAYLIST_INCOMPLETE_ON_LOAD, ("First playlist: Incomplete M3U8 data."), ("%s", skippy_m3u8_client_get_current_raw_data (demux->client)));
+      goto error;
+      break;
+    case PLAYLIST_INVALID_UTF_CONTENT:
+      GST_OBJECT_UNLOCK (demux);
+      GST_ELEMENT_ERROR (demux, SKIPPY_HLS, PLAYLIST_INVALID_UTF_CONTENT, ("First playlist: Invalid M3U8 data (buffer=%p)", demux->playlist), (NULL));
+      goto error;
+      break;
+    case NO_ERROR:
+    default:
```
Comment (tchakabam) [inline src/skippy_hlsdemux.c]: you should assert here if you reach default case here, as it means the function returns an unhandled error code. at least that would help future devs to make sure they implement handling here for other return codes of this function.

**Comment by tchakabam** [comment]
apart from that small "measure of precaution" :+1:

**Pull request merged**